source/shared/core_results.cpp

```diff
@@ -81,78 +89,231 @@ SQLPOINTER read_lob_field( sqlsrv_stmt* stmt, SQLUSMALLINT field_index, sqlsrv_b
 // dtor for each row in the cache
 void cache_row_dtor(zval* data);
 
+size_t get_float_precision(SQLLEN buffer_length, size_t unitsize)
+{
+    SQLSRV_ASSERT(unitsize != 0, "Invalid unit size!");
+
+    // get to display size by removing the null terminator from buffer length
+    size_t display_size = (buffer_length - unitsize) / unitsize;
+
+    // use the display size to determine the sql type. And if it is a double, set the precision accordingly
+    // the display sizes are set by the ODBC driver based on the precision of the sql type
+    // otherwise we can just use the default precision 
+    size_t real_display_size = 14;
+    size_t float_display_size = 24;
+    size_t real_precision = 7;
+	size_t float_precision = 15;
+
+    // For more information about display sizes for REAL vs FLOAT/DOUBLE: https://msdn.microsoft.com/en-us/library/ms713974(v=vs.85).aspx
+    // For more information about precision: https://msdn.microsoft.com/en-us/library/ms173773.aspx
+
+    // this is the case of sql type float(24) or real
+    if ( display_size == real_display_size ) {
+         return real_precision;
+    }
+    // this is the case of sql type float(53)
+    else if ( display_size == float_display_size ) {
+         return float_precision;
+    }
+
+    return 0;
+}
+
+#ifndef _WIN32
+// copy the number into a char string using the num_put facet
+template <typename Number>
+SQLRETURN get_string_from_stream(Number number_data, std::basic_string<char> &str_num, size_t precision, sqlsrv_error_auto_ptr& last_error)
+{
+    //std::locale loc( std::locale(""), new std::num_put<char> );	// By default, SQL Server doesn't take user's locale into consideration
```

Comment:
yes @david-puglielli I put it there to explain our decision back then. In case this default behavior of SQL Server is changed in the future, we could 'revert' our code.